src/app/api/applications/[id]/submit/route.ts

```diff
@@ -0,0 +1,31 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { submitApplication } from '@/services/daam/index.server';
+import { ExtendedSession, authOptions } from '@/utils/auth';
+import axios from 'axios';
+import { getServerSession } from 'next-auth';
+import { NextResponse } from 'next/server';
+
+export async function POST(request: Request, params: { params: { id: string } }) {
+  const session: ExtendedSession | null = await getServerSession(authOptions);
+  const { id } = params.params;
+
+  if (!session) {
+    return NextResponse.json({ error: 'Unauthorized' }, { status: 401 });
+  }
+
+  try {
+    submitApplication(id, session);
+    return NextResponse.json({ success: true }, { status: 200 });
+  } catch (error) {
+    if (axios.isAxiosError(error)) {
+      return NextResponse.json({ error: error.response?.data }, { status: error.response?.status });
+    } else if (error instanceof Error) {
```

Comment:
will it not always go to this else if (error instanceof Error) if it's not axios error?

Reply:
Indeed, I'm using fetch for this endpoint, so it will never end up being an AxiosError. I planned to move to axios but actually not now so I will modify the error check